File: trip-kit/src/main/java/com/skedgo/android/tripkit/HttpClientModule.java
package com.skedgo.android.tripkit;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;

@Module
public class HttpClientModule {
  @Provides public OkHttpClient.Builder httpClientBuilder() {
    return new OkHttpClient.Builder();
  }
}

File: TripKitAndroid/src/main/java/com/skedgo/android/tripkit/HttpClientModule.kt
package com.skedgo.android.tripkit

import dagger.Lazy
import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient
import skedgo.tripkit.configuration.GetAppVersion
import java.util.*

@Module
open class HttpClientModule {
  @Provides open fun httpClientBuilder(): OkHttpClient.Builder
      = OkHttpClient.Builder()

  @Provides internal fun addCustomHeaders(
      getAppVersion: GetAppVersion,
      uuidProviderLazy: Lazy<UuidProvider>,
      configs: Configs
  ): AddCustomHeaders
      = AddCustomHeaders(
      getAppVersion,
      { Locale.getDefault() },
      if (configs.isUuidOptedOut) null else uuidProviderLazy.get(),
      configs.userTokenProvider(),
      { configs.key().call() })
}
